devito/ir/clusters/algorithms.py

```diff
@@ -449,16 +449,21 @@ def callback(self, clusters, prefix, seen=None):
         return processed
 
 
+def update_reductions(processed, reductions):
```

Comment:
nitpicking, but in this case I think it should be a private function inside reduction_comms itself, that is

```
def reduction_comms(clusters):
     processed = []
     fifo = []

     def _update(reductions):
           <implementation>

     for c in clusters:
           ...
```

Reply:
That's fine, will do at rebase after the caching PR